### PR TITLE
Bugfix FXIOS-7240 [v121] Run BrowserVC activation code during app launch

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7240)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16057)

## :bulb: Description

Adds additional hooks to ensure that a few critical codepaths within BrowserVC's `appDidBecomeActive` code are run during app launch; currently that code is not run during startup, only once the app has already been launched, backgrounded, and then foregrounded.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
cc @yoanarios @OrlaM Wanted to be sure to get your 👀 on this since it impacts app launch and you have a lot more context/familiarity with this than me. So far in testing everything is looking Ok but if anything here raises any concerns please LMK. The EventQueue was not strictly needed here AFAICS but I added a related event since this ticket was part of the Event Queue epic.

**Review by adudenamedruby — APPROVED** [review]
LGTM, with the context I have for this work.

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController.swift:425]: I think with this change we can remove the call to `startAtHomeCheck` under `tabManagerDidRestoreTabs` which was added in #16805, to patch that the start at home wasn't called at start up time.

**Review by yoanarios — APPROVED** [review]
LGTM 🏅

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (mattreaganmozilla) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: Thanks @lmarceau. It looks like that startAtHomeCheck is now storing the result into `didStartAtHome` which is then checked in BVC in `urlBarPresentCFR`:

```swift
// Checks if the app started with the Homepage tab
        // If true, don't show the CFR to avoid showing it on the Homepage
        guard !didStartAtHome else {
            didStartAtHome = false
            return
        }
```

At a glance it looks like I should be able to simply move that state (`didStartAtHome`) directly into BVC with this refactor, but I will take a closer look at how these are working together.

Comment (mattreaganmozilla) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: @lmarceau Made this change in e06b285a7c3dc62c41415be4b84d3f7625f99af2. @PARAIPAN9 It looks like `didStartAtHome` was added for [FXIOS-7555](https://mozilla-hub.atlassian.net/browse/FXIOS-7555), can you LMK the best way to regression test that so I can be sure these refactors aren't impacting any of the CFR behaviors? Thank you

Comment (lmarceau) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: Yeah they had to patch something related to the CFR. There's also another fix related to the same CFR in a PR opened [here](https://github.com/mozilla-mobile/firefox-ios/pull/17172). So I'll tag @PARAIPAN9 here for review. Whatever we merge first will affect the other PR.

Comment (lmarceau) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: Yeah they had to patch something related to the CFR. There's also another fix related to the same CFR in a PR opened [here](https://github.com/mozilla-mobile/firefox-ios/pull/17095). So I'll tag @PARAIPAN9 here for review. Whatever we merge first will affect the other PR. 

Edit: I tagged the wrong PR

Comment (lmarceau) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: Mhh. Thinking, since #17095 is needed for 120, better to merge that one first then we can adjust the behavior for v121 and adjust where we can 🤔

Comment (mattreaganmozilla) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: > Thinking, since https://github.com/mozilla-mobile/firefox-ios/pull/17095 is needed for 120, better to merge that one first

@lmarceau @PARAIPAN9 I've added a Do Not Merge label for now, will wait until #17095 is merged in and then I can update this refactor to accommodate those changes.

Comment (lmarceau) [reply Client/Frontend/Browser/BrowserViewController.swift:425]: I merged it 👍 Let's work to improve this area. Let me know if I can do anything to help

Comment (mattreaganmozilla) [reply Client/Frontend/Browser/BrowserViewController.swift:428]: @lmarceau @PARAIPAN9 Update: I've rebased and updated this to incorporate Sorin's changes from #17095. Any extra 👀 are appreciated (I don't have a ton of context on the CFR fixes).

Comment (lmarceau) [reply Client/Frontend/Browser/BrowserViewController.swift:428]: I don't have full context either, so tagging Sorin on the PR for review on those changes.

Comment (PARAIPAN9) [reply Client/Frontend/Browser/BrowserViewController.swift:428]: Just switch to this branch, and started reviewing it.

Comment (PARAIPAN9) [reply Client/Frontend/Browser/BrowserViewController.swift:428]: I've test it without didStartAtHome check and it worked. I think it should be good.

**Review by lmarceau — APPROVED** [review]

**Review by PARAIPAN9 — APPROVED** [review]

**Pull request merged**
